character.cpp

```diff
@@ -33,10 +38,10 @@ void Character::attackEnemy(Character& enemy) {
 	}
 	else {
 		enemy.HP -= Character::DMG;
-		Character::XP = Character::XP + Character::DMG;
-		if (Character::XP >= 100) {
-			Character::levelUp();
-		}
+	}
+	Character::XP = Character::XP + Character::DMG;
+	if (Character::XP >= 100) {
```

Comment:
Mivel ez ki lett szedve az else ágból, most mindenképp a teljes potenciális sebzésnek megfelelő XP-t kapja, akkor is, ha ténylegesen kevesebbet sebzett. Ez így nem jó.